framework/src/libtorch/userobjects/TorchScriptUserObject.C

```diff
@@ -0,0 +1,57 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifdef LIBTORCH_ENABLED
+
+#include "TorchScriptUserObject.h"
+
+registerMooseObject("MooseApp", TorchScriptUserObject);
+
+InputParameters
+TorchScriptUserObject::validParams()
+{
+  InputParameters params = GeneralUserObject::validParams();
+  params.addClassDescription("User-facing object which loads a torch script module.");
+  params.addRequiredParam<std::string>("filename",
+                                       "The file name which contains the torch script module.");
+  params.declareControllable("filename");
+  params.addParam<bool>(
+      "load_during_construction",
+      false,
+      "If we want to load this neural network while we are constructing this object.");
+  return params;
+}
+
+TorchScriptUserObject::TorchScriptUserObject(const InputParameters & parameters)
+  : GeneralUserObject(parameters),
+    _filename(getParam<std::string>("filename")),
+    _torchscript_module(getParam<bool>("load_during_construction")
+                            ? std::make_shared<Moose::TorchScriptModule>(_filename)
+                            : nullptr)
+{
+}
+
+void
+TorchScriptUserObject::execute()
+{
+  // We load when the user executes this user object
+  _torchscript_module = std::make_shared<Moose::TorchScriptModule>(_filename);
```

Comment:
Guard this with `if (!_torchscript_module)`

Reply:
Okay I see what the use case is now. I think there are two ways forward then
1. Add a boolean option to allow "reloading" the script whenever the userobject is executed.
2. Never reload. Then implement another object deriving from this one which always reloads.

Either way is fine with me. If we go with the first approach, I'd suggest renaming the option to "reload" or "always_reload" which might be slightly more intuitive to users.

Reply:
Or how about an option:
```
addParam<ExecuteFlag>(reload_module, default(will be never), "Something");
```

And then:
```
if (_reload_flags.find(_current_execute_flag) != _reload_flags.end())
  reloadthemodule;
```

Reply:
You know what, I will just override the `execute_on` flags in here to never. And then the user can control the reload frequency through the `executo_on` flags.